dokka-subprojects/plugin-templating/src/main/kotlin/org/jetbrains/dokka/templates/DirectiveBasedTemplateProcessing.kt

```diff
@@ -41,7 +41,8 @@ public class DirectiveBasedHtmlTemplateProcessingStrategy(private val context: D
                 handleCommandAsComment(command, bodyTrimed, input, output)
             }
 
-            Files.write(output.toPath(), listOf(document.outerHtml()))
+            // drop last additional \n character that's added
+            Files.write(output.toPath(), listOf(document.outerHtml().dropLast(1)))
```

Comment:
It might not be correct to drop just the last character here, as theoretically it can be a wrong character (not a new line).

The real issue here is the wrong usage of `Files.write`, which accepts a list of lines. However, we provide the whole HTML in one line, so inside `Files.write`, it appends an additional new line at the end.
So probably better fix would be to not use `Files.write` but do something like:

```suggestion
            output.writeText(document.outerHtml())
```

Reply:
Ah, yes! Great observation. This should resolve the real issue. Thank you!